com.unity.shadergraph/Editor/Drawing/MaterialGraphEditWindow.cs

```diff
@@ -877,6 +877,7 @@ public void Initialize(string assetGuid)
                 UpdateTitle();
 
                 Repaint();
+                EditorApplication.wantsToQuit += PromptSaveIfDirtyOnQuit;
```

Comment:
Best practice is to subtract the delegate before adding to prevent duplicate registration, unless you're entirely sure Initialize will never be called twice.

Also, does this work with multiple windows open? I'm not sure if delegate registration retains the `this` context or if the second window will override the first?

Reply:
good catch on the -= : while it is in the psudo constructor method, there is a case where it gets called again without destruction. While the function is safe with the null check, best practice will have me add it to the top of initialize function

Tested with two open undocked windows, both passed save call. Since this does not reference a static method, instance method should be called. Confirmed from dotnet programming guide
>_When a delegate is constructed to wrap an instance method, the delegate references both the instance and the method. A delegate has no knowledge of the instance type aside from the method it wraps, so a delegate can refer to any type of object as long as there is a method on that object that matches the delegate signature. When a delegate is constructed to wrap a static method, it only references the method.